src/execute.rs

```diff
@@ -66,6 +72,35 @@ use std::sync::atomic::{AtomicBool, AtomicI64, Ordering};
 #[cfg(not(target_arch = "wasm32"))]
 use std::time::Duration;
 use thiserror::Error;
+
+#[cfg(not(target_arch = "wasm32"))]
+static _SOLC_REQUIREMENT: OnceLock<bool> = OnceLock::new();
+#[cfg(not(target_arch = "wasm32"))]
+fn check_solc_requirement() {
+    info!("checking solc installation..");
+    _SOLC_REQUIREMENT.get_or_init(|| {
```

Comment:
Added `solc` check into `OnceLock` for a threadsafe way of running this only once per exec.

@alexander-camuto